include/bluetooth/audio/has.h

```diff
@@ -0,0 +1,147 @@
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @brief The preset record contains all the information about a hearing aid preset. */
+struct bt_has_preset {
+	/** @brief Preset set active callback
+	 *
+	 *  The callback can also be used locally in which case no connection will be set.
+	 *  Once the preset has changed, the bt_has_active_preset_changed stall be called
+	 *  to complete the procedure.
+	 *
+	 *  @param conn   The connection that is requesting to read
+	 *  @param preset The preset that's being set active
+	 *  @param sync   Whether the server must relay this change to the other member of
+	 * 				  the Binaural Hearing Aid Set.
+	 */
+	void (*set_active)(struct bt_conn *conn, const struct bt_has_preset *preset,
+		 bool sync);
+
+	/** @brief Preset name read callback
+	 *
+	 *  The callback can also be used locally in which case no connection will be set.
+	 *
+	 *  @param conn   The connection that is requesting to read
+	 *  @param preset The preset that's name being read
+	 *  @param buf    Buffer to place the read result in
+	 *  @param len    Length of data to read
+	 *
+	 *  @return Number fo bytes read.
+	 */
+	ssize_t (*read_name)(struct bt_conn *conn, const struct bt_has_preset *preset,
+			 void *buf, uint16_t len);
+
+	/** @brief Preset name write callback
+	 * 
+	 *  If not set, the preset name is not writtable.
+	 *
+	 *  @param conn   The connection that is requesting to write
+	 *  @param preset The preset that's name being written
+	 *  @param buf    Buffer with the data to write
+	 *  @param len    Number of bytes in the buffer
+	 *
+	 *  @return Number of bytes written.
+	 */
+	ssize_t	(*write_name)(struct bt_conn *conn, const struct bt_has_preset *preset,
+			 const void *buf, uint16_t len);
+
+	/** Preset user data */
+	void *user_data;
+	/** Preset index */
+	uint8_t index;
+	/** Preset properties */
+	uint8_t properties;
+};
+
+/** @brief HAS structure */
+struct bt_has_presets {
+	/** Hearing Aid Presets */
+	const struct bt_has_preset *presets;
+	/** Hearing Aid Preset count */
+	size_t preset_count;
+};
+
+/** @def BT_HAS_PRESETS_DEFINE
+ *  @brief Statically define and register presets.
+ *
+ *  Helper macro to statically define and register presets.
+ *
+ *  @param _name Preset name.
+ */
+#define BT_HAS_PRESETS_DEFINE(_name, ...)							\
+	const struct bt_has_preset presets_##_name[] = { __VA_ARGS__ };	\
+	const Z_STRUCT_SECTION_ITERABLE(bt_has_presets, _name) = 		\
+						BT_HAS_PRESETS(presets_##_name)
+
+/** @def BT_HAS_PRESETS
+ *  @brief Hearing Aid Presets Declaration Macro.
+ *
+ *  Helper macro to declare a presets structure.
+ *
+ *  @param _presets Service attributes.
+ */
+#define BT_HAS_PRESETS(_presets)				\
+{												\
+	.presets = _presets,						\
+	.preset_count = ARRAY_SIZE(_presets),		\
+}
+
+/** @brief Make preset record unavailable for a client.
+ *
+ *  @param preset The preset that's name being disabled
+ */
+void bt_has_preset_disable(struct bt_has_preset *preset);
```

Comment:
Unrelated to this PR really, but I really dislike the acronym `has` for Hearing Aid Service as it make all functions sound like boolean functions :P

Reply:
Agree